File: articles/moral-psychology-propaganda-and-political-polarization/go/moral_exclusion_score.go
// Synthetic moral exclusion score example.
// Educational scaffold only.
//
// Run:
//   go run go/moral_exclusion_score.go

package main

import (
	"fmt"
	"math"
)

func logistic(x float64) float64 {
	return 1.0 / (1.0 + math.Exp(-x))
}

func main() {
	threat := 1.0
	propaganda := 0.8
	outrage := 0.7
	identity := 0.6
	democraticTrust := 0.4
	contact := 0.3

	latent := 0.35*threat +
		0.30*propaganda +
		0.25*outrage +
		0.20*identity -
		0.25*democraticTrust -
		0.20*contact

	risk := logistic(latent)

	fmt.Printf("Synthetic moral exclusion risk: %.3f\n", risk)
}

File: articles/moral-psychology-propaganda-and-political-polarization/cpp/moral_exclusion_score.cpp
/*
Synthetic moral exclusion score example.
Educational scaffold only.

Compile:
  c++ -std=c++17 cpp/moral_exclusion_score.cpp -o outputs/tables/moral_exclusion_score_cpp
Run:
  ./outputs/tables/moral_exclusion_score_cpp
*/

#include <cmath>
#include <iostream>

double logistic(double x) {
    return 1.0 / (1.0 + std::exp(-x));
}

int main() {
    double threat = 1.0;
    double propaganda = 0.8;
    double outrage = 0.7;
    double identity = 0.6;
    double democratic_trust = 0.4;
    double contact = 0.3;

    double latent =
        0.35 * threat +
        0.30 * propaganda +
        0.25 * outrage +
        0.20 * identity -
        0.25 * democratic_trust -
        0.20 * contact;

    double risk = logistic(latent);

    std::cout << "Synthetic moral exclusion risk: " << risk << std::endl;
    return 0;
}
